buildSrc/src/main/groovy/org/elasticsearch/gradle/test/RestIntegTestTask.groovy

```diff
@@ -95,6 +91,8 @@ class RestIntegTestTask extends DefaultTask {
          * We bypass this by instead passing this system properties vi a CommandLineArgumentProvider. This has the added
          * side-effect that these properties are NOT treated as inputs, therefore they don't influence things like the
          * build cache key or up to date checking.
+         *
+         * TODO: replace with ESTest.systemProperty
```

Comment:
If we have to choose between a slightly less concise DSL and subclassing Gradle core types then I'd choose the former. We can make this DSL better. Use something like `lazySystemProperty 'name', { 'dynamic value' }`. Let's not go strait to the nuclear option here.